atc/exec/task_config_source_test.go

```diff
@@ -38,11 +40,11 @@ var _ = Describe("TaskConfigSource", func() {
 					"a":               "b",
 					"evaluated-value": "((task-variable-name))",
 				},
-				Params: &atc.Params{
+				Params: atc.Params{
```

Comment:
Curious about this change. I think previously we would use a pointer value as we need to interpolate vars in (()) and then rewrite the value into the map.

Reply:
I think the interpolation logic is stateless now, and it returns an entirely new `TaskConfig` object rather updating nested pointers. It looks like it might be possible to update the address on that top-level pointer `TaskConfig *ImageResource` but in `task_step` and in the test, it looks like interpolation is performed, and the new value stored as a new variable that gets passed into later function calls. I don't see any evidence of `TaskConfig.ImageResource` being traversed for its inner values to be changed anymore though!

Reply:
thanks for letting us know